### DISCUSSION TITLE
Clarion and Many-to-Many Broadcast (aka blockchains)

### DISCUSSION THREAD

**Discussion by bytemaster** [bytemaster/clarion#17] [Ideas]
Your twitter account represent a personal broadcast station that only you can publish on. Other people can subscribe and any number of people can listen. There is no need to worry about abuse of other people attempting to broadcast through your channel. You can view your twitter account as a radio station where you control what is on the air 24-7.

An open chatroom that allows anyone to join is an example of a many-to-many broadcast channel. This is like a radio station that allows anyone to call in and talk at any time. The station has a limited amount of bandwidth so people will end up shouting over each other and spamming advertisements. Most people will end up tuning out of this radio station and won't bother attempting to call in.

If you put a moderator panel to screen calls then the result is a governance problem and potential censorship. You still need a means to allow anyone to broadcast to everyone without having to ask permission of anyone.

A proof of work blockchain is a radio station where who ever shouts the loudest (by burning the most electricity) gets to talk. The ability to operate a mining operation is beyond most people and the infrastructure cost is prohibitive for someone who only needs to get out a single message. Furthermore, the person shouting (aka mining blocks) is not compensating the people who have to listen (those consuming blocks).  Finally, someone who corners the market on mining hardware/cheap electricity can maintain their power forever. 

There are certain types of engagement that benefit from allowing *anyone* to broadcast to *everyone* so that consensus can be reached. An example would be registering globally unique names / profiles or implementing a community currency.

To this end Clarion will integrate an optional blockchain framework that utilizes a new consensus algorithm based upon proof of burn.  Under a proof of burn blockchain the chain with the smallest token supply (because it has burned the most) is considered the best chain. You can think of it as a blockchain with a negative block reward.  The theory is that "the power to broadcast" has value and that there is a cost to broadcast on everyone who has to listen. Burning tokens transfers value from the person broadcasting to everyone else who is waiting to broadcast. In order to maintain a monopoly over the broadcast channel you would have to have an infinite supply of money and transfer it to everyone else. As tokens become scarce their value goes up which means the "cost to broadcast" in terms of number of tokens go down. The supply never runs out because it is infinitely divisible. 

Running a BTC, ETH, or EOS node is too resource intensive for most people to run at home. This means that they are not effective broadcast tools and that most users end up dependent upon 3rd party service providers and API nodes. 

This is why Clarion proof of burn blockchains will target the bandwidth equivalent of 1 transaction per second with 1 block per minute. You could consider this a single radio station, a low-bit-rate stream.  Many applications only need this low bit rate. If an application needs more, then it could distribute itself over multiple radio stations. The more stations a user must subscribe to, the fewer users there will be operating a full node and the more centralized it will become. 

Furthermore, proof of burn blockchains don't do any validation of the data that is broadcast on the channel. This would be subject to deterministic evaluation of the receivers. This makes block validation at the consensus level very light weight and only needing to focus on token transfers necessary to implement the consensus algorithm and burn.

Proof of burn blockchains could also adopt "big blocks". This would be a radio station for power-users and companies not for end users.

### What about the tokens?

ClarionOS will not depend upon tokens and will be functional for most users without having to touch any blockchain and any tokens. The full nodes will be able to subscribe to MANY proof of burn blockchains and the blocks will propagate over the friend to friend network. Each chain will have its OWN token.  There will be no "one token to rule them all", each token will represent the economic right to broadcast on a particular channel. The creator of the channel will get to determine the distribution of the tokens. Tokens could be air dropped on EOS, BTC, BTS, and HIVE or any other chain for which a set of public keys and balances can be defined. Alternatively the channel creator could keep the tokens and then sell them to people.  My role will be to create tools, the community will have to figure out who gets the tokens and what channels to subscribe to.

### When Tokens? 

Any-to-All broadcasts will be part of phase 2 which could be a year or more away from even beginning development.

**Comment by miron** [comment]
Lets say I operate a full node for family and buddies and issue a community token which has a value of 0. Now someone outside the community wants to speak to that community (but won’t be friended by that community), buys the community token (with a different token like EOS, other established community tokens, or fiat gateway), then burns the token to talk to my community and the community gains in value. How will the different tokens be distributed inside the community?

**Reply by bytemaster** [reply]
Each community can pick its own way to distribute tokens. It could be via airdrop, it could be the founder distributing to friends, it could be based upon reputation (such as twitter followers).  There is no single / centralized token distribution model.

**Comment by jbuck84** [comment]
"Under a proof of burn blockchain the chain with the smallest token supply (because it has burned the most) is considered the best chain."

So every proof of burn blockchain that Clarion users can access will have to have standards by which the chains are launched, ie, the cost of tokens to start the network as well as the supply?

I'm a bit lost, because it seems you are implying Clarion users can access any types of chains and so I don't get how "the chain with the smallest token supply (because it has burned the most) is considered the best chain" will necessarily be true.

**Reply by bytemaster** [reply]
the "fork" with the most burned tokens... is like the BTC chain with the greatest difficulty.

**Comment by Dexaran** [comment]
# Potential message replacement

## The problem

"Forking" conversations has an inherent problem - a message or image that has already been propagated cannot be unseen even if the chain adopts a different branch.

## Example

Imagine that we have a trading channel where important financial advices are being broadcasted. Here we have three users Alice, Bob and Carol.

The system should look like this:

![3](https://user-images.githubusercontent.com/26142412/110595975-29ba0100-8198-11eb-8024-1134e81196cd.png)

Alice is welcoming others with a "Hi" message. The message is propagated and the balance of Alice decreases (she has 1 token burned in order to "fund" the message). Other nodes accept the message and the system now looks like this:

![5](https://user-images.githubusercontent.com/26142412/110596192-70a7f680-8198-11eb-91b2-e76bcd2cb4e2.png)

Carol is a malicious character here. Carol creates two versions (aka "forks") of the conversation by signing two different messages and broadcasting the first message "Buy BTC" to Bob and the second message "Buy ETH" to Alice.

Carol has only 1 token to fund just 1 message. Now the system should look like this:

![6](https://user-images.githubusercontent.com/26142412/110596797-2d9a5300-8199-11eb-957e-2930157ac804.png)

Both Alice and Bob accept their version of the message. Carol does not violate any consensus rules because her actions are legit at both versions of the chain despite the fact that these are now two different chains.

If Bob will tell Alice that the new message from Carol was broadcasted then there are two possible scenarios: (1) Alice may refuse to adopt it because in Alices version of Carol has 0 tokens already. (2) Otherwise if one of the messages is "better" than the other according to some consensus criteria (such as lower timestamp or something else) then Alice will adopt the new message replacing the "Buy ETH" with "Buy BTC".

Here comes an inherent problem - while it is possible to revert the history of messages it is not always possible to revert some events that has already happened outside of the chain.

![7](https://user-images.githubusercontent.com/26142412/110597476-098b4180-819a-11eb-9925-554dc44e20d5.png)

In classic blockchains, this problem is solved by a certain number of confirmations required for a transaction to reach "finality". However someone can still "inject" his version of the chain by investing into it.

In case of Clarion someone can broadcast messages to his version of the conversations and thus increase the value of "burned" tokens.

Imagine Bob responding with "OK" message to Carol's advice. Now the Bob's version of the conversation has burned more tokens and this is the "valid" chain since now on.

![8](https://user-images.githubusercontent.com/26142412/110600558-7eac4600-819d-11eb-93cf-5ec178a4cfe7.png)


Another problem - Carol paid for only one message, while Alice saw two messages. The message cannot be unseen even if you rewrite history, unlike a financial transaction, which transfers a value that can be returned if you revert the history.

**Reply by bytemaster** [reply]
Thanks for this analysis.  You have identified the core challenge with proof of burn implementation... the "double spending" of block production. I have intentionally not disclosed the details of how this is resolved, but rest assured that I have sufficient confidence in my solution to this problem.

I will disclose some of it now:
Every single "key" is like its own blockchain and every signature carries with it a corresponding sequence number representing the number of times a particular key has been used to sign a message.

If any node sees two messages with the same sequence number for the same key then that "key" is burned and all tokens it controls are invalid.  If Carol attempts to "fork" she takes the risk that Alice and Bob exchange notes and BURN her tokens instead and her intended transaction will have no effect.

Add in a little bit of confirmation time to allow for propagation and "fraud proofs" and you eliminate any profit from attempted double spends.

**Reply by qimiaoguo** [reply]
what do you think about mina?
blockchain seems a best solution to solve double spending
I didn't see any solution of distributed system can solve it
